src/server.rs

```diff
@@ -73,25 +73,13 @@ pub(crate) fn get_child_version<'a>(
         });
     }
 
-    // If the requested parentVersionId is the nil UUID ..
-    if parent_version_id == NIL_VERSION_ID {
-        return Ok(match client.snapshot {
-            // ..and snapshotVersionId is nil, the response is _not-found_ (the client has no
-            // versions).
-            None => GetVersionResult::NotFound,
-            // ..and snapshotVersionId is not nil, the response is _gone_ (the first version has
-            // been deleted).
-            Some(_) => GetVersionResult::Gone,
-        });
-    }
-
-    // If a version with versionId equal to the requested parentVersionId exists, the response is _not-found_ (the client is up-to-date)
-    if txn.get_version(client_id, parent_version_id)?.is_some() {
-        return Ok(GetVersionResult::NotFound);
-    }
-
-    // Otherwise, the response is _gone_ (the requested version has been deleted).
-    Ok(GetVersionResult::Gone)
+    // If the version does not exist, because the requested parent version is the latest version,
+    // then the result is NotFound. Otherwise, that version is Gone..
```

Comment:
```suggestion
    // If the version does not exist and the requested parent version is the latest version or,
    // the latest version is nil, then the result is NotFound. Otherwise, that version is Gone.
```

Word it how you like, but the combination of the *because* and the ambiguity of English tripped me up here and made me think you were saying that the version doesn't exist *because* `client.latest_version_id == parent_version_id`, which i couldn't make any sense of because that is normally the case when a version *does* exist and we return `GetVersionResult::Success`, right?

Reply:
Thanks, I appreciate the comment -- I reworded it.